pulsar-client/src/main/java/org/apache/pulsar/client/impl/UnAckedMessageTracker.java

```diff
@@ -138,8 +138,11 @@ public void run(Timeout t) throws Exception {
                         if (!headPartition.isEmpty()) {
                             log.info("[{}] {} messages will be re-delivered", consumerBase, headPartition.size());
                             headPartition.forEach(messageId -> {
-                                addChunkedMessageIdsAndRemoveFromSequenceMap(messageId, messageIds, consumerBase);
-                                messageIds.add(messageId);
+                                if (messageId instanceof ChunkMessageIdImpl) {
+                                    addChunkedMessageIdsAndRemoveFromSequenceMap(messageId, messageIds, consumerBase);
+                                } else {
+                                    messageIds.add(messageId);
```

Comment:
Because `messageIds` is a `Set`, the change here is an improvement to save once `set.add()`. The original code does not have bugs, right?

Reply:
And could you add a test to ensure that the chunked message info can be removed from the `UnAckedMessageTracker` after the chunked messages are acknowledged or negatively acknowledged? I didn't find any similar tests, so adding a test to ensure all things work as expected is better. Thanks

Reply:
Added a new test `testTrackChunkedMessageId` to cover it.